src/main/kotlin/com/sourcegraph/cody/agent/protocol/WorkspaceEditOperation.kt

```diff
@@ -0,0 +1,18 @@
+package com.sourcegraph.cody.agent.protocol
+
+data class WorkspaceEditParamsOptions(
+    val overwrite: Boolean = false,
+    val ignoreIfNotExists: Boolean = false,
+    val recursive: Boolean = false
+)
+
+data class WorkspaceEditOperation(
+    val type: String, // all
+    val uri: String? = null, // created, delete, edit
```

Comment:
Can we make sure URI parameters are sent as `URI`s and not strings?
Generally I think we should be extra careful around uri/path objects, they were source of many conversion problems in paths between VSCode <-> Intellij, especially adding additional platforms as windows.

Reply:
They have been bouncing back and forth in origin/main between URI and String and causing me no end of merge headaches. My understanding is that URI has trouble serializing in some cases.

It works now, and I don't want to break it, so let's plan an API cleanup pass afterwards.